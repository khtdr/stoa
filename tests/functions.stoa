
#!/usr/bin/env stoa

fun makeCounter(start) {
  var n = start;
  return fun() {
    n = n+1;
    return n;
  };
}

// chained calls?
print makeCounter(212)();

// need to peek ahead?
fun() {};
