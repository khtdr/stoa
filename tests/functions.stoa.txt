213
102,102
107,103,1
done
