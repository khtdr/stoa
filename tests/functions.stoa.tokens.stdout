FUN[3,1]
IDENTIFIER[3,5]
LEFT_PAREN[3,16]
IDENTIFIER[3,17]
RIGHT_PAREN[3,22]
LEFT_CURL[3,24]
VAR[4,3]
IDENTIFIER[4,7]
EQUAL[4,9]
IDENTIFIER[4,11]
SEMICOLON[4,16]
RETURN[5,3]
FUN[5,10]
LEFT_PAREN[5,13]
RIGHT_PAREN[5,14]
LEFT_CURL[5,16]
IDENTIFIER[6,5]
EQUAL[6,7]
IDENTIFIER[6,9]
PLUS[6,10]
NUMBER[6,11]
SEMICOLON[6,12]
RETURN[7,5]
IDENTIFIER[7,12]
SEMICOLON[7,13]
RIGHT_CURL[8,3]
SEMICOLON[8,4]
RIGHT_CURL[9,1]
PRINT[11,1]
IDENTIFIER[11,7]
LEFT_PAREN[11,18]
NUMBER[11,19]
RIGHT_PAREN[11,22]
LEFT_PAREN[11,23]
RIGHT_PAREN[11,24]
SEMICOLON[11,25]
FUN[13,1]
LEFT_PAREN[13,4]
RIGHT_PAREN[13,5]
LEFT_CURL[13,7]
RIGHT_CURL[13,8]
SEMICOLON[13,9]
FUN[15,1]
IDENTIFIER[15,5]
LEFT_PAREN[15,16]
IDENTIFIER[15,17]
RIGHT_PAREN[15,24]
LEFT_CURL[15,26]
RETURN[16,3]
IDENTIFIER[16,10]
LEFT_PAREN[16,21]
IDENTIFIER[16,22]
LEFT_PAREN[16,29]
RIGHT_PAREN[16,30]
RIGHT_PAREN[16,31]
SEMICOLON[16,32]
RIGHT_CURL[17,1]
FUN[19,1]
IDENTIFIER[19,5]
LEFT_PAREN[19,16]
IDENTIFIER[19,17]
RIGHT_PAREN[19,24]
LEFT_CURL[19,26]
VAR[20,3]
IDENTIFIER[20,7]
EQUAL[20,9]
IDENTIFIER[20,11]
LEFT_PAREN[20,22]
IDENTIFIER[20,23]
LEFT_PAREN[20,30]
RIGHT_PAREN[20,31]
RIGHT_PAREN[20,32]
SEMICOLON[20,33]
RETURN[21,3]
FUN[21,10]
LEFT_PAREN[21,13]
RIGHT_PAREN[21,14]
LEFT_CURL[21,16]
VAR[22,5]
IDENTIFIER[22,9]
EQUAL[22,12]
IDENTIFIER[22,14]
LEFT_PAREN[22,21]
RIGHT_PAREN[22,22]
SEMICOLON[22,23]
VAR[23,5]
IDENTIFIER[23,9]
EQUAL[23,12]
IDENTIFIER[23,14]
LEFT_PAREN[23,15]
RIGHT_PAREN[23,16]
SEMICOLON[23,17]
RETURN[24,5]
IDENTIFIER[24,12]
DASH[24,15]
IDENTIFIER[24,17]
SEMICOLON[24,19]
RIGHT_CURL[25,3]
SEMICOLON[25,4]
RIGHT_CURL[26,1]
VAR[28,1]
IDENTIFIER[28,5]
EQUAL[28,7]
IDENTIFIER[28,9]
LEFT_PAREN[28,20]
NUMBER[28,21]
RIGHT_PAREN[28,24]
SEMICOLON[28,25]
VAR[29,1]
IDENTIFIER[29,5]
EQUAL[29,7]
IDENTIFIER[29,9]
LEFT_PAREN[29,20]
IDENTIFIER[29,21]
RIGHT_PAREN[29,22]
SEMICOLON[29,23]
PRINT[31,1]
IDENTIFIER[31,7]
LEFT_PAREN[31,8]
RIGHT_PAREN[31,9]
PLUS[31,11]
STRING[31,13]
PLUS[31,17]
IDENTIFIER[31,19]
LEFT_PAREN[31,20]
RIGHT_PAREN[31,21]
SEMICOLON[31,22]
VAR[33,1]
IDENTIFIER[33,5]
EQUAL[33,7]
IDENTIFIER[33,9]
LEFT_PAREN[33,20]
IDENTIFIER[33,21]
RIGHT_PAREN[33,22]
SEMICOLON[33,23]
IDENTIFIER[35,1]
LEFT_PAREN[35,2]
RIGHT_PAREN[35,3]
SEMICOLON[35,4]
IDENTIFIER[35,6]
LEFT_PAREN[35,7]
RIGHT_PAREN[35,8]
SEMICOLON[35,9]
IDENTIFIER[35,11]
LEFT_PAREN[35,12]
RIGHT_PAREN[35,13]
SEMICOLON[35,14]
PRINT[37,1]
IDENTIFIER[37,7]
LEFT_PAREN[37,8]
RIGHT_PAREN[37,9]
PLUS[37,11]
STRING[37,13]
PLUS[37,17]
IDENTIFIER[37,19]
LEFT_PAREN[37,20]
RIGHT_PAREN[37,21]
PLUS[37,23]
STRING[37,25]
PLUS[37,29]
IDENTIFIER[37,31]
LEFT_PAREN[37,32]
RIGHT_PAREN[37,33]
SEMICOLON[37,34]
PRINT[39,1]
STRING[39,7]
SEMICOLON[39,13]
